xbmc/platform/darwin/ios/XBMCController.mm

```diff
@@ -760,8 +760,19 @@ - (bool) presentFramebuffer
 //--------------------------------------------------------------
 - (CGSize) getScreenSize
 {
-  screensize.width  = m_glView.bounds.size.width * screenScale;
-  screensize.height = m_glView.bounds.size.height * screenScale;
+  __block CGSize tmp;
+  if ([NSThread isMainThread]) {
+    tmp.width  = m_glView.bounds.size.width * screenScale;
+    tmp.height = m_glView.bounds.size.height * screenScale;
```

Comment:
Because I am unsure if I can set _block on a member of the class (which screensize is) - _block needs to be set to allow the variable to be altered by another run loop in a block.

Reply:
correct